### PR TITLE
[CP-stable]SelectionArea's selection should not be cleared on loss of window focus

### PR DESCRIPTION
**Pull request by flutteractionsbot**
This pull request is created by [automatic cherry pick workflow](https://github.com/flutter/flutter/blob/main/docs/releases/Flutter-Cherrypick-Process.md#automatically-creates-a-cherry-pick-request)
Please fill in the form below, and a flutter domain expert will evaluate this cherry pick request.

### Issue Link:
What is the link to the issue this cherry-pick is addressing?

There was no issue created (can create an issue if required).

PR description:
```
This change fixes an issue where SelectionArea would clear its selection when the application window lost focus by first checking if the application is running. This is needed because FocusManager is aware of the application lifecycle as of https://github.com/flutter/flutter/pull/142930 , and triggers a focus lost if the application is not active.

Also fixes an issue where the FocusManager was not being reset on tests at the right time, causing it always to build with TargetPlatform.android as its context.
```

### Changelog Description:
Explain this cherry pick in one line that is accessible to most Flutter developers. See [best practices](https://github.com/flutter/flutter/blob/main/docs/releases/Hotfix-Documentation-Best-Practices.md) for examples

The selection of a `SelectionArea` or `SelectableRegion` should not be lost when the window loses focus.

### Impact Description:
What is the impact (ex. visual jank on Samsung phones, app crash, cannot ship an iOS app)? Does it impact development (ex. flutter doctor crashes when Android Studio is installed), or the shipping production app (the app crashes on launch)

This mainly impacts desktop devices that can click out of the current window or tab. When clicking out of the window the user expects their selection will still be there when they return. Without this change the selection is lost when the window loses focused. On mobile devices the FocusManager is not applifecycle aware.

### Workaround:
Is there a workaround for this issue?

No workaround since the FocusManager being aware of the app lifecycle is an implementation detail introduced in https://github.com/flutter/flutter/pull/142930.

### Risk:
What is the risk level of this cherry-pick?

  - [x] Low
  - [ ] Medium
  - [ ] High

### Test Coverage:
Are you confident that your fix is well-tested by automated tests?

  - [x] Yes
  - [ ] No

### Validation Steps:
What are the steps to validate that this fix works?

When running the example in the `SelectionArea` docs https://api.flutter.dev/flutter/material/SelectionArea-class.html, the selection should be retained when the window loses focus.

Steps:
1. Select some text.
2. Click to another tab or window.
3. Come back to previous tab or window.
4. Selection should be retained.

### CODE REVIEW

**Review by nate-thegrate — APPROVED** [review]
This cherry-pick unblocks a fix for a regression that affects text input on iOS (see #151144).

**Comment by chunhtai** [comment]
@Renzo-Olivares do you know what is the state of this CP?

**Comment by Renzo-Olivares** [comment]
@chunhtai I think this is still waiting for a review from the release team.

**Comment by DylanCross** [comment]
Thanks for working this guys. For a datapoint, our login autofill has been broken on dev since like April. I'd really like to get a flutter version with this fixed ASAP. Thanks.

**Comment by itsjustkevin** [comment]
@Renzo-Olivares this branch is no longer alive.  I will close this in favor of a new cherry-pick to the 3.24 branch if it is still necessary.

**Comment by zSoNz** [comment]
Hello! The issues related to this bug are still occurring on version 3.24.


https://github.com/user-attachments/assets/c863f1cf-1575-454f-b7c1-6c0d34d69f7c

**Comment by nate-thegrate** [comment]
@zSoNz That's unfortunate to hear.

But I don't think it pertains to this pull request, since this PR is trying to cherry-pick a commit that is present in 3.24 into 3.22.

**Comment by zSoNz** [comment]
> @zSoNz That's unfortunate to hear.
> 
> But I don't think it pertains to this pull request, since this PR is trying to cherry-pick a commit that is present in 3.24 into 3.22.

Got you, thanks.

**Pull request closed without merging**
